sympy/solvers/solveset.py

```diff
@@ -108,6 +108,9 @@ def _invert(f_x, y, x, domain=S.Complexes):
     if not isinstance(s, FiniteSet) or x1 == f_x:
```

Comment:
I guess the condition `or x1 != x` could be appended here instead of adding a new block below.

Reply:
Thanks. You are right. Modification done.